relentless/ensemble.py

```diff
@@ -8,132 +12,273 @@
 from . import core
 
 class RDF(core.Interpolator):
+    """Constructs a :py:class:`Interpolator` object interpolating through an RDF
+    function :math:`g(r)`, and creates an `nx2` array with columns as `r` and `g`.
+
+    Parameters
+    ----------
+    r : array_like
+        1-d array of r values, that must be continually increasing.
+    g : array_like
+        1-d array of g values.
+
+    """
     def __init__(self, r, g):
         super().__init__(r,g)
         self.table = np.column_stack((r,g))
 
+class Volume(abc.ABC):
+    """Abstract base class for a defined volume/region."""
+    @property
+    @abc.abstractmethod
+    def volume(self):
+        """Volume of the region."""
+        pass
+
+class Parallelepiped(Volume):
+    """General triclinic box.
+
+    Parameters
+    ----------
+    a : array_like
+        One of the vectors forming the parallelepiped; must be an array
+        containing elements as (`x`,`y`,`z`).
+    b : array_like
+        One of the vectors forming the parallelepiped; must be an array
+        containing elements as (`x`,`y`,`z`).
+    c : array_like
+        One of the vectors forming the parallelepiped; must be an array
+        containing elements as (`x`,`y`,`z`).
+
+    Raises
+    ------
+    TypeError
+        If a, b, and c are not all `3x1` arrays.
+
+    """
+    def __init__(self, a, b, c):
+        self.a = np.asarray(a)
```

Comment:
I think we should add `dtype=np.float64` to these so that we get double precision. It isn't strictly necessary, but lower precision could cause issues with the cross / dot products in really skewed boxes.